editor/store/selectors.js

```diff
@@ -1196,152 +1203,280 @@ export function getNotices( state ) {
 }
 
 /**
- * An item that appears in the inserter. Inserting this item will create a new
- * block. Inserter items encapsulate both regular blocks and shared blocks.
- *
- * @typedef {Object} Editor.InserterItem
- * @property {string}   id                Unique identifier for the item.
- * @property {string}   name              The type of block to create.
- * @property {Object}   initialAttributes Attributes to pass to the newly created block.
- * @property {string}   title             Title of the item, as it appears in the inserter.
- * @property {string}   icon              Dashicon for the item, as it appears in the inserter.
- * @property {string}   category          Block category that the item is associated with.
- * @property {string[]} keywords          Keywords that can be searched to find this item.
- * @property {boolean}  isDisabled        Whether or not the user should be prevented from inserting this item.
- */
-
-/**
- * Given a regular block type, constructs an item that appears in the inserter.
+ * Determines if the given block type is allowed to be inserted, and, if
+ * parentUID is provided, whether it is allowed to be nested within the given
+ * parent.
  *
- * @param {Object}           state             Global application state.
- * @param {string[]|boolean} allowedBlockTypes Allowed block types, or true/false to enable/disable all types.
- * @param {Object}           blockType         Block type, likely from getBlockType().
+ * @param {Object} state      Global application state.
+ * @param {string} blockName  The name of the given block type, e.g. 'core/paragraph'.
+ * @param {?string} parentUID The parent that the given block is to be nested within, or null.
  *
- * @return {Editor.InserterItem} Item that appears in inserter.
+ * @return {boolean} Whether or not the given block type is allowed to be inserted.
  */
-function buildInserterItemFromBlockType( state, allowedBlockTypes, blockType ) {
-	if ( ! allowedBlockTypes || ! blockType ) {
-		return null;
-	}
+export const canInsertBlockType = createSelector(
+	( state, blockName, parentUID = null ) => {
```

Comment:
Why do we assign a default value for `parentUID` ?

Reply:
>So that one can call `wp.data.select( 'core/editor' ).canInsertBlockType( 'block-id' )` to determine if a block is insertable into the post.

My concern is less that it's optional, more about whether `null` is treated specially here. In other words, would anything change if `= null` were removed?

Reply:
Ah, got it. Nothing would change—`undefined` would work just as well. I tend use ` = null` as a way of explicitly denoting which parameters are optional. This may be because I'm not used to working in codebases with consistent JSDOc comments 🙂